### ISSUE TITLE
Healthvault lib zip file curl is broken, if building your own Docker image

### ISSUE THREAD

**Issue by jaredsieling** [openmhealth/shimmer#22]
If you are building your own docker image (option 3), line 35 of the Dockerfile uses curl to fetch the Healthvault Java Lib.  However, the URL is no longer valid, and returns an HTML file sending you to:

http://healthvaultjavalib.codeplex.com/downloads/get/878881

If you try the curl to that address, it still fails.  A similar issue was discussed [here](https://groups.google.com/d/msg/chocolatey/tQJzs0B7a1k/-r59Peefbe8J), but I was not able to get that to work.

As a work around, I just downloaded the .zip file to the /docker/source directory, and then had the Dockerfile copy it to the container while building:

```
COPY ./healthvault-sdk-1.6.0.zip /root/healthvault-sdk-1.6.0.zip
```

**Comment by emersonf** [comment]
Thanks for the heads up @jaredsieling. It looks like MS released a new internal build, so I've updated the HealthVault library link and it should work again.

On a related note, since we wrote those Docker instructions, we've become more proficient with Docker and Docker Compose (previously called Fig) has been gaining ground. We're looking into simplifying deployments using Compose, so I'm curious if/why people are using the single source Dockerfile to know if it's worth keeping around. Could you let us know why you prefer that method over the binary build? And whether a straightforward Compose build would be easier?

**Comment by jaredsieling** [comment]
Thanks. We've made some updates to the shim code for our purposes and added another API, so I am packaging it with those updates.  A straightforward Compose build would be great, and I actually was looking into that myself. It would be great to connect about that, or if you have an early alpha version of that you could make that available.

**Comment by emersonf** [comment]
Compose builds are available as of 0.3.0, in two flavours. The default `docker-compose.yml` configuration downloads images, whereas the `docker-compose-build.yml` builds the images. When using the latter configuration, we recommend using the convenience script called `run-dockerized.sh` that compiles the code before delegating to Docker Compose.

PS. If you'd like to contribute the API you added, let us know.
